frontend/fingertips-frontend/lib/fetchData.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  Configuration,
+  WeatherForecast,
+  WeatherForecastApi,
+} from '@/generated-sources/api-client';
+import { connection } from 'next/server';
+
+export async function fetchData(): Promise<WeatherForecast[]> {
+  // We don't want to render this page statically
+  await connection();
```

Comment:
I think this `connection()` call should stay in the page, as it's about the behaviour of the page, and not necessarily something we always want to call when this function is called.